File: app/src/main/java/com/zaclimon/tsutaeru/ui/settings/TsutaeruSettingsFragment.java
package com.zaclimon.tsutaeru.ui.settings;

import android.app.Activity;
import android.support.v17.leanback.widget.ArrayObjectAdapter;

import com.zaclimon.xipl.ui.settings.ProviderSettingsTvFragment;

/**
 * Fragment handling the Settings from a TV standpoint. Concrete {@link ProviderSettingsTvFragment}
 * for Tsutaeru.
 *
 * @author zaclimon
 * Creation date: 01/07/17
 */

public class TsutaeruSettingsFragment extends ProviderSettingsTvFragment {

    @Override
    public Class<? extends Activity> getSettingsElementActivity() {
        return (TsutaeruSettingsElementActivity.class);
    }

    @Override
    public ArrayObjectAdapter getSettingsObjectAdapter() {
        return (new TsutaeruSettingsObjectAdapter());
    }

}

File: app/src/main/java/com/zaclimon/tsutaeru/ui/settings/TsutaeruSettingsFragment.kt
package com.zaclimon.tsutaeru.ui.settings

import android.app.Activity
import android.support.v17.leanback.widget.ArrayObjectAdapter
import com.zaclimon.xipl.ui.settings.ProviderSettingsTvFragment

/**
 * Fragment handling the Settings from a TV standpoint. Concrete implementation of a
 * [ProviderSettingsTvFragment] for Tsutaeru.
 *
 * @author zaclimon
 */
class TsutaeruSettingsFragment : ProviderSettingsTvFragment() {

    override fun getSettingsElementActivity(): Class<out Activity> {
        return (TsutaeruSettingsElementActivity::class.java)
    }

    override fun getSettingsObjectAdapter(): ArrayObjectAdapter {
        return (TsutaeruSettingsObjectAdapter())
    }

}